### DISCUSSION TITLE
CustomSearch for select / autosuggest

### DISCUSSION THREAD

**Discussion by HavardPede** [fremtind/jokul#2403] [Annet]
Ved utvikling av ny tilbudsflyt i SAM så vi nødvendigheten for en ny funksjonalitet i SAM. Vi ønsket å bruke select komponenten for å kunne velge mellom kunder i tilbudet. Men noen ganger trengte vi kunder som ikke allerede lå i tilbudet, og som dermed ikke lå som et valg i selecten. 
Vi tok inspirasjon fra bla. søkefunksjonaliteten i Teams og kom frem til denne løsningen: 

![image](https://user-images.githubusercontent.com/30959515/138255726-ffadbaeb-a1f1-4868-9ba9-cebeda13a302.png)
![image](https://user-images.githubusercontent.com/30959515/138255800-e30719b2-f121-439e-8808-0be9b634cf4a.png)

Den er basert på Select komponenten. Her får du da muligheten til å bruke en custom funksjonalitet for å søke etter verdier som ikke ligger foreslått i selecten. 

Er dette en funksjonalitet som ønskes som standard funksjonalitet enten på select, autosuggest eller lignende? 

(Inspirasjon tatt fra Teams søkefunksjonalitet)
![image](https://user-images.githubusercontent.com/30959515/138256207-380babbc-5d9f-482f-ae01-29a83c391dbe.png)

**Comment by kennidenni** [comment]
Har dere mulighet til å søke opp basert på navn eller må dere søke opp med personnummer slik som vi gjør i vår løsning? Ikke at det løser akkurat dette problemet, men jeg bare lurer. 
I en av våre løsninger har vi input for personnummer og så kommer det opp et inputfelt med navn under som blir preutfylt med infoen som vi får fra backend når vi søker opp personnummeret.

**Reply by HavardPede** [reply]
@kennidenni Vi har en implementasjon for kundesøk som både støtter navn og ssn. Dette kommer opp som en modal. 
Denne blir bla. brukt for å legge til personer som skal forsikres (personforsikring). Dette er via en knapp, typ "Legg til person". 

Senere skal du sette info slik som betaler osv. Disse er veldig ofte de samme personene som er lagt til i tilbudet tidligere. Dermed ønsker vi å foreslå disse personene (fks via en select). I de tilfellene der du ønsker å velge en person som ikke ligger på tilbudet fra før, ønsker vi da å åpne kundesøk modalen. Dermed ble implementasjonen ovenfor foreslått.
